src/main.ts

```diff
@@ -0,0 +1,41 @@
+import * as core from '@actions/core';
+import * as installer from './installer';
+import * as auth from './authutil';
+import * as path from 'path';
+
+export async function run() {
+  try {
+    //
+    // Version is optional.  If supplied, install / use from the tool cache
+    // If not supplied then task is still used to setup proxy, auth, etc...
+    //
+    let version = core.getInput('node-version');
+    if (!version) {
+      version = core.getInput('version');
```

Comment:
even though this is officially deprecated, I left it in because I wasn't in the mood to break folks as part of these larger changes.